[Automation] native-image run fails for org.xerial:sqlite-jdbc:3.40.0.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.40.0.0" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.39.2.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25260953949/job/74068131006
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
 and  35,912 methods found reachable
    2,707 types,     275 fields, and   1,700 methods registered for reflection
       71 types,      92 fields, and     289 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.0s @ 0.98GB)
[4/8] Parsing methods...      [**]                                                                       (3.9s @ 0.65GB)
[5/8] Inlining methods...     [***]                                                                      (2.0s @ 0.71GB)
[6/8] Compiling methods...    [****]                                                                    (17.6s @ 0.83GB)
[7/8] Laying out methods...   [**]                                                                       (3.2s @ 0.80GB)
[8/8] Creating image...       [**]                                                                       (3.2s @ 0.94GB)
   9.54MB (18.98%) for code area:    22,372 compilation units
  36.90MB (73.41%) for image heap:  214,631 objects and 332 resources
   3.82MB ( 7.61%) for other data
  50.26MB in total image size, 48.24MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.56MB java.base                                           24.12MB byte[] for general heap data
   2.15MB svm.jar (Native Image)                               2.75MB byte[] for code metadata
 558.98kB assertj-core-3.22.0.jar                              2.57MB byte[] for java.lang.String
 481.25kB java.xml                                             1.37MB java.lang.String
 313.41kB sqlite-jdbc-3.40.0.0.jar                             1.02MB java.lang.Class
 207.98kB junit-jupiter-engine-5.9.2.jar                     850.72kB com.oracle.svm.core.hub.DynamicHubCompanion
 120.76kB junit-platform-launcher-1.9.2.jar                  549.73kB heap alignment
 103.55kB java.logging                                       449.41kB java.util.concurrent.ConcurrentHashMap$Node
  99.40kB junit-platform-engine-1.9.2.jar                    304.24kB java.lang.Object[]
  81.79kB junit-4.13.2.jar                                   301.73kB java.util.HashMap$Node
 517.39kB for 25 more packages                                 2.61MB for 2052 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (897B). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        4.9s (6.5% of total time) in 668 GCs | Peak RSS: 1.75GB | CPU load: 3.62
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.xerial.sqlite-jdbc_tests-tests' in 1m 15s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.sqlite.SQLiteJDBCLoader in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

Failed to load native library:sqlite-3.40.0.0-9e29429e-6092-4541-bc14-0b51ba9cf6f3-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/sqlite-3.40.0.0-9e29429e-6092-4541-bc14-0b51ba9cf6f3-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:105)
	at java.base@25.0.3/java.lang.ClassLoader.loadLibrary(ClassLoader.java:143)
	at java.base@25.0.3/java.lang.Runtime.load0(Runtime.java:767)
	at java.base@25.0.3/java.lang.System.load(System.java:1647)
	at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
	at org.sqlite.SQLiteJDBCLoader.extractAndLoadLibraryFile(SQLiteJDBCLoader.java:237)
	at org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:350)
	at org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
	at org.sqlite.core.NativeDB.load(NativeDB.java:63)
	at org.sqlite.SQLiteConnection.open(SQLiteConnection.java:278)
	at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
	at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
	at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
	at org.sqlite.JDBC.createConnection(JDBC.java:104)
	at org.sqlite.SQLiteDataSource.getConnection(SQLiteDataSource.java:440)
	at org.sqlite.SQLiteDataSource.getConnection(SQLiteDataSource.java:432)
	at org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest.createsDatabaseMetadataAndReadsDriverDetails(JDBC3DatabaseMetaDataTest.java:23)
	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.3/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
Failed to load native library:sqlite-3.40.0.0-4878d473-ba3b-4f0f-9dfb-a3d95c281b9c-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/junit5418427271477401148/sqlite-3.40.0.0-4878d473-ba3b-4f0f-9dfb-a3d95c281b9c-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:105)
	at java.base@25.0.3/java.lang.ClassLoader.loadLibrary(ClassLoader.java:143)
	at java.base@25.0.3/java.lang.Runtime.load0(Runtime.java:767)
	at java.base@25.0.3/java.lang.System.load(System.java:1647)
	at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
	at org.sqlite.SQLiteJDBCLoader.extractAndLoadLibraryFile(SQLiteJDBCLoader.java:237)
	at org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:350)
	at org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
	at org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest.initializeExtractsPackagedNativeLibraryResource(SQLiteJDBCLoaderTest.java:28)
	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.3/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest > createsDatabaseMetadataAndReadsDriverDetails() FAILED

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenExplicitPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenReferenceClassPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteConnectionTest > opensDatabaseFromClasspathResource() FAILED

org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest > initializeExtractsPackagedNativeLibraryResource() FAILED

org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test > pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection() FAILED


Failures (4):
  JUnit Jupiter:JDBC3DatabaseMetaDataTest:createsDatabaseMetadataAndReadsDriverDetails()
    MethodSource [className = 'org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest', methodName = 'createsDatabaseMetadataAndReadsDriverDetails', methodParameterTypes = '']
    => java.sql.SQLException: Error opening connection
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:281)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]
     Caused by: java.lang.Exception: No native library found for os.name=Linux, os.arch=x86_64, paths=[/org/sqlite/native/Linux/x86_64:/usr/lib64:/lib64:/lib:/usr/lib]
       org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:375)
       org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
       org.sqlite.core.NativeDB.load(NativeDB.java:63)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:278)
       [...]
  JUnit Jupiter:SQLiteConnectionTest:opensDatabaseFromClasspathResource()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLiteConnectionTest', methodName = 'opensDatabaseFromClasspathResource', methodParameterTypes = '']
    => java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:41)
       org.sqlite.core.NativeDB._open_utf8(Native Method)
       org.sqlite.core.NativeDB._open(NativeDB.java:75)
       org.sqlite.core.DB.open(DB.java:216)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]
  JUnit Jupiter:SQLiteJDBCLoaderTest:initializeExtractsPackagedNativeLibraryResource()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest', methodName = 'initializeExtractsPackagedNativeLibraryResource', methodParameterTypes = '']
    => java.lang.Exception: No native library found for os.name=Linux, os.arch=x86_64, paths=[/org/sqlite/native/Linux/x86_64:/usr/lib64:/lib64:/lib:/usr/lib]
       org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:375)
       org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
       org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest.initializeExtractsPackagedNativeLibraryResource(SQLiteJDBCLoaderTest.java:28)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       [...]
  JUnit Jupiter:SQLitePooledConnectionAnonymous1Test:pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test', methodName = 'pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection', methodParameterTypes = '']
    => java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:41)
       org.sqlite.core.NativeDB._open_utf8(Native Method)
       org.sqlite.core.NativeDB._open(NativeDB.java:75)
       org.sqlite.core.DB.open(DB.java:216)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]

Test run finished after 96 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[         6 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         4 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 18s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.xerial:sqlite-jdbc:3.39.2.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 20s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.40.0.0 failed with exit code 1.
FAILED[native-image run][3.40.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.40.0.0" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.39.2.1"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.40.0.0" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.39.2.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25260953949/job/74068131024
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
*]                                                                   (18.7s @ 0.79GiB)
[7/8] Laying out methods...   [**]                                                                      (3.5s @ 0.82GiB)
[8/8] Creating image...       [**]                                                                      (3.4s @ 0.96GiB)
   8.44MiB (17.84%) for code area:    22,654 compilation units
  35.19MiB (74.35%) for image heap:  219,295 objects and 200 resources
   3.70MiB ( 7.82%) for other data
  47.33MiB in total image size, 45.38MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 921.20KiB o.g.n.runtime/com.oracle.svm.core.code             22.58MiB byte[] for embedded resources
 720.92KiB java.base/java.util                                 2.58MiB byte[] for code metadata
 408.35KiB java.base/jdk.internal.classfile.impl               2.54MiB byte[] for string data
 324.00KiB java.base/java.lang                                 1.38MiB java.base/java.lang.String
 242.44KiB java.base/java.util.concurrent                      0.99MiB java.base/java.lang.Class
 228.84KiB java.base/java.text                               908.69KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 224.55KiB java.base/com.sun.crypto.provider                 563.27KiB byte[] for general heap data
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       322.07KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 155.64KiB java.base/java.io                                 309.77KiB byte[] for reflection metadata
 149.52KiB java.base/java.lang.invoke                        300.03KiB java.base/java.util.HashMap$Node
   4.93MiB for 317 more packages                               2.77MiB for 2065 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (975B). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.2s (6.4% of total time) in 755 GCs | Peak RSS: 1.63GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests (executable, 45.38MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.xerial.sqlite-jdbc_tests-tests' in 1m 20s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest > createsDatabaseMetadataAndReadsDriverDetails() FAILED

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenExplicitPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenReferenceClassPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteConnectionTest > opensDatabaseFromClasspathResource() FAILED

org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest > initializeExtractsPackagedNativeLibraryResource() FAILED

org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test > pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection() FAILED


Failures (4):
  JUnit Jupiter:JDBC3DatabaseMetaDataTest:createsDatabaseMetadataAndReadsDriverDetails()
    MethodSource [className = 'org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest', methodName = 'createsDatabaseMetadataAndReadsDriverDetails', methodParameterTypes = '']
    => java.sql.SQLException: Error opening connection
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:281)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]
     Caused by: java.lang.Exception: No native library found for os.name=Linux, os.arch=x86_64, paths=[/org/sqlite/native/Linux/x86_64:/usr/lib64:/lib64:/lib:/usr/lib]
       org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:375)
       org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
       org.sqlite.core.NativeDB.load(NativeDB.java:63)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:278)
       [...]
  JUnit Jupiter:SQLiteConnectionTest:opensDatabaseFromClasspathResource()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLiteConnectionTest', methodName = 'opensDatabaseFromClasspathResource', methodParameterTypes = '']
    => java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:41)
       org.sqlite.core.NativeDB._open_utf8(Native Method)
       org.sqlite.core.NativeDB._open(NativeDB.java:75)
       org.sqlite.core.DB.open(DB.java:216)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]
  JUnit Jupiter:SQLiteJDBCLoaderTest:initializeExtractsPackagedNativeLibraryResource()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest', methodName = 'initializeExtractsPackagedNativeLibraryResource', methodParameterTypes = '']
    => java.lang.Exception: No native library found for os.name=Linux, os.arch=x86_64, paths=[/org/sqlite/native/Linux/x86_64:/usr/lib64:/lib64:/lib:/usr/lib]
WARNING: A restricted method in java.lang.System has been called
       org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:375)
WARNING: java.lang.System::load has been called by org.sqlite.SQLiteJDBCLoader in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

Failed to load native library:sqlite-3.40.0.0-6bafafd0-dd8f-4f57-86e6-bbc134e471a2-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/sqlite-3.40.0.0-6bafafd0-dd8f-4f57-86e6-bbc134e471a2-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:105)
	at java.base@25.0.2/java.lang.ClassLoader.loadLibrary(ClassLoader.java:166)
	at java.base@25.0.2/java.lang.Runtime.load0(Runtime.java:767)
	at java.base@25.0.2/java.lang.System.load(System.java:1646)
	at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
	at org.sqlite.SQLiteJDBCLoader.extractAndLoadLibraryFile(SQLiteJDBCLoader.java:237)
	at org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:350)
	at org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
	at org.sqlite.core.NativeDB.load(NativeDB.java:63)
	at org.sqlite.SQLiteConnection.open(SQLiteConnection.java:278)
	at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
	at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
	at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
	at org.sqlite.JDBC.createConnection(JDBC.java:104)
	at org.sqlite.SQLiteDataSource.getConnection(SQLiteDataSource.java:440)
	at org.sqlite.SQLiteDataSource.getConnection(SQLiteDataSource.java:432)
	at org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest.createsDatabaseMetadataAndReadsDriverDetails(JDBC3DatabaseMetaDataTest.java:23)
	at java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.2/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Failed to load native library:sqlite-3.40.0.0-17615509-fe72-4271-bf33-1c9461103014-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/junit12399830785751131964/sqlite-3.40.0.0-17615509-fe72-4271-bf33-1c9461103014-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:105)
	at java.base@25.0.2/java.lang.ClassLoader.loadLibrary(ClassLoader.java:166)
	at java.base@25.0.2/java.lang.Runtime.load0(Runtime.java:767)
	at java.base@25.0.2/java.lang.System.load(System.java:1646)
	at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
	at org.sqlite.SQLiteJDBCLoader.extractAndLoadLibraryFile(SQLiteJDBCLoader.java:237)
	at org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:350)
	at org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
	at org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest.initializeExtractsPackagedNativeLibraryResource(SQLiteJDBCLoaderTest.java:28)
	at java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.2/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.
       org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
       org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest.initializeExtractsPackagedNativeLibraryResource(SQLiteJDBCLoaderTest.java:28)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       [...]
  JUnit Jupiter:SQLitePooledConnectionAnonymous1Test:pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test', methodName = 'pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection', methodParameterTypes = '']
    => java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:41)
       org.sqlite.core.NativeDB._open_utf8(Native Method)
       org.sqlite.core.NativeDB._open(NativeDB.java:75)
       org.sqlite.core.DB.open(DB.java:216)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]

Test run finished after 41 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[         6 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         4 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

BUILD FAILED in 1m 24s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.xerial:sqlite-jdbc:3.39.2.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 26s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.40.0.0 failed with exit code 1.
FAILED[native-image run][3.40.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.40.0.0" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.39.2.1"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.40.0.0" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.39.2.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25260953949/job/74068131052
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
.9s @ 0.83GiB)
[6/8] Compiling methods...    [*****]                                                                  (21.8s @ 0.86GiB)
[7/8] Laying out methods...   [**]                                                                      (3.5s @ 1.04GiB)
[8/8] Creating image...       [**]                                                                      (4.0s @ 1.22GiB)
   9.54MiB (18.71%) for code area:    24,936 compilation units
  37.50MiB (73.53%) for image heap:  256,742 objects and 202 resources
   3.96MiB ( 7.76%) for other data
  51.00MiB in total image size, 48.88MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.03MiB o.g.n.runtime/com.oracle.svm.core.code             22.58MiB byte[] for embedded resources
 728.08KiB java.base/java.util                                 2.93MiB byte[] for code metadata
 408.35KiB java.base/jdk.internal.classfile.impl               2.82MiB byte[] for string data
 332.77KiB java.base/java.lang                                 1.51MiB java.base/java.lang.String
 267.01KiB java.base/com.sun.crypto.provider                   1.06MiB java.base/java.lang.Class
 242.65KiB java.base/java.util.concurrent                    966.44KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 231.39KiB java.base/java.text                               922.10KiB byte[] for general heap data
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       464.73KiB heap alignment
 192.84KiB java.base/sun.security.provider                   389.07KiB java.base/java.lang.Object[]
 173.16KiB java.base/sun.security.x509                       377.76KiB byte[] for reflection metadata
   5.78MiB for 327 more packages                               3.55MiB for 2210 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (977B). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       7.3s (8.2% of total time) in 765 GCs | Peak RSS: 1.88GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests (executable, 48.88MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.xerial.sqlite-jdbc_tests-tests' in 1m 28s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile

WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.sqlite.SQLiteJDBCLoader in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

Failed to load native library:sqlite-3.40.0.0-cefc9021-9eff-4709-a7f0-47a793c1f64d-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/sqlite-3.40.0.0-cefc9021-9eff-4709-a7f0-47a793c1f64d-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:105)
	at java.base@25.0.2/java.lang.ClassLoader.loadLibrary(ClassLoader.java:166)
	at java.base@25.0.2/java.lang.Runtime.load0(Runtime.java:767)
	at java.base@25.0.2/java.lang.System.load(System.java:1646)
	at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
	at org.sqlite.SQLiteJDBCLoader.extractAndLoadLibraryFile(SQLiteJDBCLoader.java:237)
	at org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:350)
	at org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
	at org.sqlite.core.NativeDB.load(NativeDB.java:63)
	at org.sqlite.SQLiteConnection.open(SQLiteConnection.java:278)
	at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
	at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
	at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
	at org.sqlite.JDBC.createConnection(JDBC.java:104)
	at org.sqlite.SQLiteDataSource.getConnection(SQLiteDataSource.java:440)
	at org.sqlite.SQLiteDataSource.getConnection(SQLiteDataSource.java:432)
	at org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest.createsDatabaseMetadataAndReadsDriverDetails(JDBC3DatabaseMetaDataTest.java:23)
	at java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.2/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)
Failed to load native library:sqlite-3.40.0.0-ad44d5e2-cc31-46f3-8d16-d6209e67c1e2-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/junit18416653477380587666/sqlite-3.40.0.0-ad44d5e2-cc31-46f3-8d16-d6209e67c1e2-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:105)
	at java.base@25.0.2/java.lang.ClassLoader.loadLibrary(ClassLoader.java:166)
	at java.base@25.0.2/java.lang.Runtime.load0(Runtime.java:767)
	at java.base@25.0.2/java.lang.System.load(System.java:1646)
	at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
	at org.sqlite.SQLiteJDBCLoader.extractAndLoadLibraryFile(SQLiteJDBCLoader.java:237)
	at org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:350)
	at org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
	at org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest.initializeExtractsPackagedNativeLibraryResource(SQLiteJDBCLoaderTest.java:28)
	at java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.2/java.lang.Thread.run(Thread.java:1474)
> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest > createsDatabaseMetadataAndReadsDriverDetails() FAILED

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenExplicitPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenReferenceClassPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteConnectionTest > opensDatabaseFromClasspathResource() FAILED

org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest > initializeExtractsPackagedNativeLibraryResource() FAILED

org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test > pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection() FAILED


Failures (4):
  JUnit Jupiter:JDBC3DatabaseMetaDataTest:createsDatabaseMetadataAndReadsDriverDetails()
    MethodSource [className = 'org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest', methodName = 'createsDatabaseMetadataAndReadsDriverDetails', methodParameterTypes = '']
    => java.sql.SQLException: Error opening connection
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:281)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]
     Caused by: java.lang.Exception: No native library found for os.name=Linux, os.arch=x86_64, paths=[/org/sqlite/native/Linux/x86_64:/usr/lib64:/lib64:/lib:/usr/lib]
       org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:375)
       org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
       org.sqlite.core.NativeDB.load(NativeDB.java:63)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:278)
       [...]
  JUnit Jupiter:SQLiteConnectionTest:opensDatabaseFromClasspathResource()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLiteConnectionTest', methodName = 'opensDatabaseFromClasspathResource', methodParameterTypes = '']
    => java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:41)
       org.sqlite.core.NativeDB._open_utf8(Native Method)
       org.sqlite.core.NativeDB._open(NativeDB.java:75)
       org.sqlite.core.DB.open(DB.java:216)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]
  JUnit Jupiter:SQLiteJDBCLoaderTest:initializeExtractsPackagedNativeLibraryResource()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest', methodName = 'initializeExtractsPackagedNativeLibraryResource', methodParameterTypes = '']
    => java.lang.Exception: No native library found for os.name=Linux, os.arch=x86_64, paths=[/org/sqlite/native/Linux/x86_64:/usr/lib64:/lib64:/lib:/usr/lib]
       org.sqlite.SQLiteJDBCLoader.loadSQLiteNativeLibrary(SQLiteJDBCLoader.java:375)
       org.sqlite.SQLiteJDBCLoader.initialize(SQLiteJDBCLoader.java:68)
       org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest.initializeExtractsPackagedNativeLibraryResource(SQLiteJDBCLoaderTest.java:28)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)

       [...]
  JUnit Jupiter:SQLitePooledConnectionAnonymous1Test:pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection()
    MethodSource [className = 'org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test', methodName = 'pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection', methodParameterTypes = '']
    => java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:41)
       org.sqlite.core.NativeDB._open_utf8(Native Method)
       org.sqlite.core.NativeDB._open(NativeDB.java:75)
       org.sqlite.core.DB.open(DB.java:216)
       org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
       org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
       org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
       org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
       org.sqlite.JDBC.createConnection(JDBC.java:104)
       [...]

Test run finished after 88 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]

[         0 containers failed     ]
[         6 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         2 tests successful      ]
[         4 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 32s
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.xerial:sqlite-jdbc:3.39.2.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 34s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.40.0.0 failed with exit code 1.
FAILED[native-image run][3.40.0.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.40.0.0" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.39.2.1"]
```

--- a/metadata/org.xerial/sqlite-jdbc/3.40.0.0/reachability-metadata.json
+++ b/metadata/org.xerial/sqlite-jdbc/3.40.0.0/reachability-metadata.json
@@ -1,0 +1,266 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "java.lang.Throwable",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "toString",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.javax.SQLitePooledConnection$1"
+      },
+      "type" : "java.sql.Connection",
+      "methods" : [
+        {
+          "name" : "createStatement",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getAutoCommit",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "isClosed",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Collation",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "xCompare",
+          "parameterTypes" : [
+            "java.lang.String",
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Function",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "args"
+        },
+        {
+          "name" : "context"
+        },
+        {
+          "name" : "value"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "xFunc",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Function$Aggregate",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "clone",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "xFinal",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "xStep",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Function$Window",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "xInverse",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "xValue",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.ProgressHandler",
+      "jniAccessible" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.core.DB",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "throwex",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "throwex",
+          "parameterTypes" : [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.core.DB$ProgressObserver",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "progress",
+          "parameterTypes" : [
+            "int",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.core.NativeDB",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "pointer"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "stringToUtf8ByteArray",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "throwex",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.core.NativeDB"
+      },
+      "type" : "org.sqlite.core.NativeDB",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "busyHandler"
+        },
+        {
+          "name" : "commitListener"
+        },
+        {
+          "name" : "progressHandler"
+        },
+        {
+          "name" : "updateListener"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "sun.security.provider.NativePRNG",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.security.SecureRandomParameters"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "sun.security.provider.SHA",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.javax.SQLitePooledConnection"
+      },
+      "type" : {
+        "proxy" : [
+          "java.sql.Connection"
+        ]
+      }
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.util.ResourceFinder"
+      },
+      "glob" : "/org_xerial/sqlite_jdbc/resource_finder_missing_resource.txt"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "glob" : "META-INF/maven/org.xerial/sqlite-jdbc/pom.properties"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "glob" : "org/sqlite/native/Linux/x86_64/libsqlitejdbc.so"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.jdbc3.JDBC3DatabaseMetaData"
+      },
+      "glob" : "sqlite-jdbc.properties"
+    }
+  ]
+}

--- a/metadata/org.xerial/sqlite-jdbc/index.json
+++ b/metadata/org.xerial/sqlite-jdbc/index.json
@@ -1,13 +1,28 @@
 [
   {
     "latest" : true,
-    "metadata-version" : "3.39.2.1",
+    "metadata-version" : "3.40.0.0",
+    "test-version" : "3.36.0.3",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/$version$/sqlite-jdbc-$version$-sources.jar",
     "repository-url" : "https://github.com/xerial/sqlite-jdbc",
     "test-code-url" : "https://github.com/xerial/sqlite-jdbc/tree/$version$/src/test",
     "documentation-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/$version$/sqlite-jdbc-$version$-javadoc.jar",
     "description" : "SQLite JDBC is a JDBC driver for accessing and creating SQLite database files from Java applications. It packages native SQLite libraries for major operating systems into a single JAR so applications can use SQLite without separate native setup.",
+    "tested-versions" : [
+      "3.40.0.0"
+    ],
+    "allowed-packages" : [
+      "org.sqlite"
+    ]
+  },
+  {
+    "metadata-version" : "3.39.2.1",
     "test-version" : "3.36.0.3",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/$version$/sqlite-jdbc-$version$-sources.jar",
+    "repository-url" : "https://github.com/xerial/sqlite-jdbc",
+    "test-code-url" : "https://github.com/xerial/sqlite-jdbc/tree/$version$/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/$version$/sqlite-jdbc-$version$-javadoc.jar",
+    "description" : "SQLite JDBC is a JDBC driver for accessing and creating SQLite database files from Java applications. It packages native SQLite libraries for major operating systems into a single JAR so applications can use SQLite without separate native setup.",
     "tested-versions" : [
       "3.39.2.1"
     ],

--- a/stats/org.xerial/sqlite-jdbc/3.40.0.0/stats.json
+++ b/stats/org.xerial/sqlite-jdbc/3.40.0.0/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "3.40.0.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 3,
+          "totalCalls" : 3
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 5,
+          "totalCalls" : 5
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 8,
+      "totalCalls" : 8
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 5746,
+        "missed" : 15635,
+        "ratio" : 0.268743,
+        "total" : 21381
+      },
+      "line" : {
+        "covered" : 1262,
+        "missed" : 3559,
+        "ratio" : 0.261771,
+        "total" : 4821
+      },
+      "method" : {
+        "covered" : 264,
+        "missed" : 1106,
+        "ratio" : 0.192701,
+        "total" : 1370
+      }
+    }
+  } ]
+}

--- a/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build.gradle
+++ b/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build.gradle
@@ -24,4 +24,14 @@ graalvmNative {
             }
         }
     }
+    binaries {
+        all {
+            // sqlite-jdbc 3.40.x loads a JNI library that expects libm symbols at runtime.
+            buildArgs.addAll([
+                '-H:NativeLinkerOption=-Wl,--no-as-needed',
+                '-H:NativeLinkerOption=-lm',
+                '-H:NativeLinkerOption=-Wl,--as-needed'
+            ])
+        }
+    }
 }

